test/route-hooks.test.js

```diff
@@ -209,6 +209,59 @@ function testBeforeHandlerHook (hook) {
     })
   })
 
+  test(`${hook} option should handle throwing objects`, t => {
+    t.plan(4)
+    const fastify = Fastify()
+
+    const myError = { myError: 'kaboom' }
+
+    fastify.setErrorHandler(async (error, request, reply) => {
+      t.deepEqual(error, myError, 'the error object throws by the user')
+      reply.send({ this: 'is', my: 'error' })
+    })
+
+    fastify.get('/', {
+      [hook]: async () => {
+        // eslint-disable-next-line no-throw-literal
+        throw myError
+      }
+    }, (req, reply) => {
+      t.fail('the handler must not be called')
+    })
+
+    fastify.inject({
+      url: '/',
+      method: 'GET'
+    }, (err, res) => {
+      t.error(err)
+      t.is(res.statusCode, 500)
+      t.deepEqual(res.json(), { this: 'is', my: 'error' })
+    })
+  })
+
+  test(`${hook} option should handle throwing objects by default`, t => {
+    t.plan(3)
+    const fastify = Fastify()
+
+    fastify.get('/', {
+      [hook]: async () => {
+        // eslint-disable-next-line no-throw-literal
+        throw { myError: 'kaboom', message: 'i am an error' }
```

Comment:
What if you were to `throw 'This is an error?'`?

Reply:
The user will get always what it throws, so it will be a plain string response with status code 500.

I will add a test for it!